.github/workflows/SSW.SophieBot.yml

```diff
@@ -0,0 +1,152 @@
+name: BuildAndDeploy
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - src/SSW.SophieBot/**
+      - .github/workflows/SSW.SophieBot.yml
+      - .github/workflows/Build-LUIS.ps1
+      - .github/workflows/Build-Orchestrator.ps1
+      - .github/workflows/LUUtils.ps1
+      - .github/workflows/PublishUtils.ps1
+      - .github/workflows/Publish-WebApp.ps1
+
+concurrency:
+  group: bot_build_and_deploy
+  cancel-in-progress: true
+
+jobs:
+  build-and-deploy:
+    name: Build and deploy
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+    env:
+      SourceDirectory: ${{ github.workspace }}
+      BotProjectDirectory: ${{ github.workspace }}/src/SSW.SophieBot/SSWSophieBot
+    steps:
+      - uses: actions/checkout@v2.3.4
+
+      # Install prerequisites
+      - name: Use Node 13.x
+        uses: actions/setup-node@v2.4.0
+        with:
+          node-version: '13'
+      - name: Use BF CLI
+        run: npm install -g @microsoft/botframework-cli
+      - name: Use .Net Core sdk 3.1.x
+        uses: actions/setup-dotnet@v1.8.1
+        with:
+          dotnet-version: '3.1.x'
+      
+      - name: Create appsettings file
+        uses: jsdaniell/create-json@1.1.2
+        with:
+          name: "appsettings.json"
+          json: ${{ secrets.APPSETTINGS_JSON }}
+          dir: "src/SSW.SophieBot/SSWSophieBot/settings"
+      
+      - name: Set environment variables
```

Comment:
You'd better explain to me what this section does

Reply:
"Create appsettings file" section is for creating appsettings.json from GitHub secret,
"Set environment variables" section is for extracting Azure keys and credentials from the publishing profile in appsettings.json into environment variables, so they can be used by later steps

Reply:
Can't later steps have environment variables respectively?

Reply:
I have done refinements for env variables settings and usages. See latest commits.